emission/net/ext_service/habitica/proxy.py

```diff
@@ -126,37 +124,37 @@ def habiticaProxy(user_uuid, method, method_url, method_args):
 
 
 def setup_party(user_id, group_id_from_url, inviterId):
-  group_id = list(edb.get_habitica_db().find({"user_id": user_id}))[0]['habitica_group_id']
-  if group_id is None:
-    #check if user is already in a party
-    try:
-      method_url = "/api/v3/user"
-      result = habiticaProxy(user_id, 'GET', method_url, None)
-      data = result.json()
-      group_id = data['data']['party']['_id']
-      edb.get_habitica_db().update({"user_id": user_id},{"$set": {'habitica_group_id': group_id}},upsert=True)
-
-    except KeyError:
-      group_id = group_id_from_url
-      invite_uri = "/api/v3/groups/"+group_id+"/invite"
-      logging.debug("invite user to party api url = %s" % invite_uri)
-      user_val = list(edb.get_habitica_db().find({"user_id": user_id}))[0]
-      method_args = {'uuids': [user_val['habitica_id']], 'inviter': group_id, 'emails': []}
-      emInviterId = edb.get_habitica_db().find_one({"habitica_id": inviterId})["user_id"]
-      response = habiticaProxy(emInviterId, 'POST', invite_uri, method_args)
-      logging.debug("invite user to party response = %s" % response)
-      join_url = "/api/v3/groups/"+group_id+"/join"
-      response2 = habiticaProxy(user_id, 'POST', join_url, {})
-      response.raise_for_status()
-      response2.raise_for_status()
-      edb.get_habitica_db().update({"user_id": user_id},{"$set": {'habitica_group_id': group_id}},upsert=True)
+  #group_id = list(edb.get_habitica_db().find({"user_id": user_id}))[0]['habitica_group_id']
```

Comment:
well, you could check it by getting the profile from habitica and seeing if there is an existing group. That's what it means by saying that habitica is the source of truth - you can still use the fields, you just have to get them from habitica or if you cache them, you have to invalidate the cache.